formula-android/src/main/java/com/instacart/formula/integration/ActivityStoreContext.kt

```diff
@@ -20,58 +18,43 @@ import io.reactivex.functions.BiFunction
  *
  * @param Activity Type of activity that this class provides context for.
  */
-class ActivityStoreContext<Activity : FragmentActivity>(
-    @PublishedApi internal val holder: ActivityHolder<Activity>
-) {
-    private val fragmentFlowStateRelay: BehaviorRelay<FragmentFlowState> = BehaviorRelay.create()
-    private val activityResultRelay: PublishRelay<ActivityResult> = PublishRelay.create()
-
-    internal fun onActivityResult(result: ActivityResult) {
-        activityResultRelay.accept(result)
-    }
+abstract class ActivityStoreContext<out Activity : FragmentActivity> {
+
+    // TODO: might be okay to remove in favor of startedActivity
+    @PublishedApi internal abstract fun currentActivity(): Activity?
+
+    @PublishedApi internal abstract fun startedActivity(): Activity?
```

Comment:
I wonder if we should just either make this public 
or not inline for easier stacks if there is an issue, trusting the [JIT + precompiler + ML perf profiles + whatever else is in the crazy android optimization pipeline] to amortize the cost, if any

Reply:
I'd like to avoid making these part of public API. However, your point about removing `inline` usage from this class might be the way to go.